### PR TITLE
Enable teuthology to run basic deepsea integration test

### PR DESCRIPTION
**Pull request by smithfarm**
for now this clones DeepSea from my fork, wip-teuthology-suite branch (and, sadly, this is hardcoded)

### CODE REVIEW

**Comment by smithfarm** [comment]
Tested here [1], works in the sense that the `basic_health_ok.sh` script is run and HEALTH_OK is reached.

[1] http://149.202.177.102:8081/ubuntu-2017-06-07_21:14:48-deepsea-test.yaml-ses5---basic-openstack/

**Comment by smithfarm** [comment]
We had spoken about using Ceph's `workunit` task, but since the `deepsea` task is already cloning the DeepSea git repo it makes more sense to let the `deepsea` task take a list of commands to run (perhaps after first changing the current working directory `DeepSea/qa/) - haven't implemented that part yet, though.

File: qa/tasks/deepsea.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        super(DeepSea, self).setup()
+
+        self.log.info("master remote: {}".format(self.config["master_remote"]))
+
+        self.salt.master_remote.run(args=[
+            'git',
+            'clone',
+            '--branch',
+            'wip-teuthology-suite',
+            'https://github.com/smithfarm/DeepSea.git',
```
Comment (smithfarm) [inline qa/tasks/deepsea.py]: This needs to be rewritten so the `deepsea` task takes `repo` and `branch` options that default to `https://github.com/SUSE/DeepSea` and `master`, respectively.

Comment (smithfarm) [inline qa/tasks/deepsea.py]: Pushed commit that implements this

File: qa/tasks/deepsea.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            'deepsea task only accepts a dict for configuration'
+        self.config["repo"] = self.config.get('repo', 'https://github.com/SUSE/DeepSea.git')
+        self.config["branch"] = self.config.get('branch', 'master')
+
+        # determine the role id of the master role
+        if(misc.num_instances_of_type(self.cluster, 'master') != 1):
+            raise ConfigError('deepsea requires a single master role')
+        for id_ in misc.all_roles_of_type(self.ctx.cluster, 'master'):
+            # this loop should iterate only once
+            master_role = '.'.join(['master', str(id_)])
```
Comment (rjfd) [inline qa/tasks/deepsea.py]: In this case, why not using just:
```
id_ = misc.all_roles_of_type(self.ctx.cluster, 'master')[0]
master_role = '.'.join(['master', str(id_)])
```

Comment (smithfarm) [inline qa/tasks/deepsea.py]: Why? Because I didn't realize the generator function could be used in that way. Fixing.

Comment (rjfd) [inline qa/tasks/deepsea.py]: ah, my bad! Didn't realize it was a generator. In that case, try this:
```
id_ = next(misc.all_roles_of_type(self.ctx.cluster, 'master'))
master_role = '.'.join(['master', str(id_)])
```

**Comment by smithfarm** [comment]
Since merging stuff into `ses5` causes teuthlogy to rebuild all the ceph RPMs, I'm going to close this, but consider it "in". From now on, let's target DeepSea suite PRs against the `wip-deepsea-suite` branch, not `ses5`.

**Comment by smithfarm** [comment]
Do not delete the branch :-)

**Pull request closed without merging**